model/src/main/java/jetbrains/jetpad/model/composite/Composites.java

```diff
@@ -269,6 +278,43 @@ int deltaBetween(CompositeT c1, CompositeT c2) {
     }
   }
 
+  /**
+   * @return Lowest common ancestor for the given nodes
+   *  or {@code null} when they have no common ancestors.
+   */
+  public static <CompositeT extends Composite<CompositeT>>
+  CompositeT commonAncestor(CompositeT object1, CompositeT object2) {
+    if (object1 == object2) {
+      return object1;
+    } else if (isDescendant(object1, object2)) {
```

Comment:
The stack traversal below wouldn't work if `object1 == object2`, descendant checks are for optimization.